Makefile

```diff
@@ -113,6 +113,69 @@ distclean:
 
 check: test
 
+# Remove files generated by running coverage, put the non-instrumented lib back 
+# in place 
+coverage-clean:
+	if [ -d lib_ ]; then rm -rf lib; mv lib_ lib; fi
+	-rm -rf gcovr
+	-rm -rf out/$(BUILDTYPE)/.coverage
+	-rm -rf .cov_tmp coverage
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/*.gcda
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/tracing/*.gcda
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/*.gcno
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/tracing*.gcno
+
+# Build and test with code coverage reporting.  Leave the lib directory 
+# instrumented for any additional runs the user may want to make.
+# For C++ coverage reporting, this needs to be run in conjunction with configure
+#  --coverage.  html coverage reports will be created under coverage/
+
+coverage: coverage-build coverage-test
+
+coverage-build: all
+	mkdir node_modules
```

Comment:
Can you make this `mkdir -p` so it doesn’t fail when `node_modules` is already there?